okhttp/src/main/java/okhttp3/internal/connection/RouteSelector.kt

```diff
@@ -95,17 +95,19 @@ class RouteSelector(
   /** Prepares the proxy servers to try. */
   private fun resetNextProxy(url: HttpUrl, proxy: Proxy?) {
     eventListener.proxySelectStart(call, url)
-    proxies = if (proxy != null) {
+    proxies = run {
```

Comment:
is it cleaner just to have an inline function than to use run and return@run?

Reply:
Perhaps? I changed to use a local function.